client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -818,6 +818,28 @@ Undocumented.prototype.getTitanMailboxAvailability = function ( domain, mailbox,
 	);
 };
 
+/**
+ * Deletes a mailbox from a Professional Email(i.e. Titan) account
+ *
+ * @param domain The domain name of the account
+ * @param mailbox The mailbox name to delete
+ * @param fn The callback function
+ */
+Undocumented.prototype.deleteTitanMailbox = function ( domain, mailbox, fn ) {
+	debug( 'DELETE: /emails/titan/:domain/mailbox/:mailbox', { domain, mailbox } );
+
+	return this.wpcom.req.get(
+		{
+			path: `/emails/titan/${ encodeURIComponent( domain ) }/mailbox/${ encodeURIComponent(
+				mailbox
+			) }`,
+			method: 'DELETE',
+			apiNamespace: 'wpcom/v2',
+		},
+		fn
+	);
+};
+
```

Comment:
Thank you for the ping, @stephanethomas !

> As an aside, I personally find the confirmation message a bit alarming (which in some ways may probably be a good thing).

I think it might make sense to be alarming as they are about to lose their mailbox and data esp if they reached here in error.  Aside from that being alarming, I think there's an opportunity to communicate what the user should expect from the removal a bit in detail here. Sharing a couple of things that came up to my mind,

- When they should expect the mailbox to be removed? Instantly or at the end of the billing cycle?
- What if they want to export their existing Inbox? Is there a way to have a backup? If yes how to do it?

I also don't know the answer to these questions. 😃  Let me know your thoughts!

Reply:
In answer to your questions @saygunnyc 
- The mailbox is removed instantly so they have no access to it. However the data is only deleted 30 days after they removed the mailbox. Within that window, it can only be re-activated manually
- The only way to backup/export their mailbox data right now is to use a third-party mail client.

Reply:
FYI @stephanethomas , I have migrated to react-query